pubtools/_pulp/tasks/common.py

```diff
@@ -158,3 +169,87 @@ def publish_with_cache_flush(self, repos, units=None):
         out.append(flush_ud)
 
         return out
+
+
+class PushItemRecorder(CollectorService):
```

Comment:
The functionality here is specifically about deleting, and the names of methods can be misleading if neither the class nor method names indicate this. e.g. nobody would guess that PushItemRecorder.push_item_for* methods always return a DELETED push item and aren't appropriate to call in other cases.  Do you think the class could be given a name which is more specifically about deletion?

Reply:
Agree. Need to modify this class for the reason you mentioned and extend this for handling of pushitems from sources other than tasks.